lib/streamlit/user_info.py

```diff
@@ -516,3 +520,38 @@ def to_dict(self) -> UserInfo:
             A dictionary of the current user's information.
         """
         return _get_user_info()
+
+
+has_shown_experimental_user_warning = False
+
+
+def maybe_show_deprecated_user_warning() -> None:
+    """Show a deprecation warning for the experimental_user alias."""
+    global has_shown_experimental_user_warning  # noqa: PLW0603
```

Comment:
I understand this is a way to only show this once. Does this mean that the deprecation notice will disappear after the first run (and is that our standard)?

Reply:
Correct, this is a way to show it only once, and it is our standard for other cases too

Similar to `deprecate_func_name` function that havily used in `__init__.py` we have `deprecate_obj_name` function in our `deprecation_utils.py`
But in this case it is inapplicable, bause the `deprecate_obj_name` wrapper functionality breaks the `st.experimental_user` functionality (e.g. the behaviour of `st.write`).

The reason why the "global" used it here instead of method used in e.g. `_create_deprecated_obj_wrapper` is in that case it would be possible to mock it and write test for it. (See [this test](https://github.com/streamlit/streamlit/pull/11095/files#diff-c9e7505993b4c1d59e38061ec771fbf5578ad8ee75afb9c9f40c083cda8ded40R148)